# **LOAD DATA**

In [ ]:
#Import modul yang dibutuhkan

import pandas as pd

In [ ]:
#Upload dataset

from google.colab import files
uploaded = files.upload()

Saving kompi_e.csv to kompi_e.csv


In [ ]:
#Load dataset

df = pd.read_csv('kompi_e.csv')
df

nomor_peserta                         nama                      jurusan  \
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
11        DS5E12              Renata Havresia                   Akuntansi    
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (cm)  
0                 79                172  
1                 49                153  
2                 41                156  
3                 56                158  
4                 60                173  
5                 69                168  
6                 57                162  
7                 49                162  
8                 58                167  
9                 50                158  
10                50                169  
11                63                159  
12                54                156  
13                45                150  
14                48                150  
15                50                162  
16                50                165  
17                58                168  
18                40                165  
19                45                160  
20                52                165  
21                54                160  
22                53                159

In [ ]:
#Cek dimensi dataframe
#Untuk mengetahui baris dan kolom

df.shape

(23, 5)

**Identifikasi Data**

Setelah data di load, menggunakan bantuan package Pandas dan function read, dapat diketahui bahwa dataset tersebut terdiri dari 23 baris dan 6 kolom.

Kolom 1, berisi indeks tabel dataframe.

Kolom 2, nomor_peserta. Berisi id peserta yang membedakan anatar peserta satu dengan peserta yang lainnya. Tipe data pada kolom ini berupa string.

Kolom 3, nama. Berisi nama peserta. Tipe data pada kolom ini berupa string.

Kolom 4, jurusan. Berisi jurusan dari peserta. Tipe data pada kolom ini berupa string.

Kolom 5, berat_badan (kg). Berisi berat badan dari peserta dalam satuan kilogram. Tipe data pada kolom ini adalah integer.

Kolom 6, tinggi_badan (cm). Berisi tinggi badan dari peserta dalam satuan sentimeter. Tipe data pada kolom ini adalah integer.

## **Program Menghitung BMI**

In [ ]:
#Mengubah kolom tinggi badan dalam satuan cm ke m

df['tinggi_badan (m)'] = df['tinggi_badan (cm)'] / 100

In [ ]:
df

nomor_peserta                         nama                      jurusan  \
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
11        DS5E12              Renata Havresia                   Akuntansi    
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (cm)  tinggi_badan (m)  
0                 79                172              1.72  
1                 49                153              1.53  
2                 41                156              1.56  
3                 56                158              1.58  
4                 60                173              1.73  
5                 69                168              1.68  
6                 57                162              1.62  
7                 49                162              1.62  
8                 58                167              1.67  
9                 50                158              1.58  
10                50                169              1.69  
11                63                159              1.59  
12                54                156              1.56  
13                45                150              1.50  
14                48                150              1.50  
15                50                162              1.62  
16                50                165              1.65  
17                58                168              1.68  
18                40                165              1.65  
19                45                160              1.60  
20                52                165              1.65  
21                54                160              1.60  
22                53                159              1.59

In [ ]:
#Menghapus kolom "tinggi_badan (cm)"
#Karena sudah tidak digunakan lagi dan dapat mengganggu proses analisis lebih lanjut

df = df.drop('tinggi_badan (cm)', axis=1)
df

nomor_peserta                         nama                      jurusan  \
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
11        DS5E12              Renata Havresia                   Akuntansi    
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (m)  
0                 79              1.72  
1                 49              1.53  
2                 41              1.56  
3                 56              1.58  
4                 60              1.73  
5                 69              1.68  
6                 57              1.62  
7                 49              1.62  
8                 58              1.67  
9                 50              1.58  
10                50              1.69  
11                63              1.59  
12                54              1.56  
13                45              1.50  
14                48              1.50  
15                50              1.62  
16                50              1.65  
17                58              1.68  
18                40              1.65  
19                45              1.60  
20                52              1.65  
21                54              1.60  
22                53              1.59

In [ ]:
#Function untuk menghitung BMI

def calculate_bmi(dataframe, weight_col, height_col):
    """
    Menghitung BMI (Indeks Massa Tubuh) dan menambahkannya sebagai kolom baru ke dalam DataFrame.

    Parameters:
    - dataframe: DataFrame yang ingin diubah.
    - weight_col: Nama kolom berat badan dalam DataFrame.
    - height_col: Nama kolom tinggi badan dalam DataFrame.

    Returns:
    - DataFrame asli dengan kolom BMI tambahan.
    """
    # Membuat salinan DataFrame agar tidak memodifikasi DataFrame asli.
    df = dataframe.copy()

    # Menghitung BMI dan menambahkannya sebagai kolom baru.
    df['BMI'] = df[weight_col] / (df[height_col] ** 2)

    return df

In [ ]:
#Memanggil dan menjalankan function yang sudah dideklarasikan

df_with_bmi = calculate_bmi(df, 'berat_badan (kg)', 'tinggi_badan (m)')

df_BMI = pd.DataFrame(df_with_bmi)
df_BMI

nomor_peserta                         nama                      jurusan  \
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
11        DS5E12              Renata Havresia                   Akuntansi    
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (m)        BMI  
0                 79              1.72  26.703624  
1                 49              1.53  20.932120  
2                 41              1.56  16.847469  
3                 56              1.58  22.432303  
4                 60              1.73  20.047446  
5                 69              1.68  24.447279  
6                 57              1.62  21.719250  
7                 49              1.62  18.670934  
8                 58              1.67  20.796730  
9                 50              1.58  20.028842  
10                50              1.69  17.506390  
11                63              1.59  24.919900  
12                54              1.56  22.189349  
13                45              1.50  20.000000  
14                48              1.50  21.333333  
15                50              1.62  19.051974  
16                50              1.65  18.365473  
17                58              1.68  20.549887  
18                40              1.65  14.692378  
19                45              1.60  17.578125  
20                52              1.65  19.100092  
21                54              1.60  21.093750  
22                53              1.59  20.964361

# **Program Memberikan Status BMI**

In [ ]:
#Function untuk memberikan tambahan status BMI pada dataframe

def add_bmi_status(dataframe):
    """
    Menambahkan kolom 'BMI Status' ke dalam DataFrame berdasarkan nilai BMI.

    Parameters:
    - dataframe: DataFrame yang ingin diubah.

    Returns:
    - DataFrame asli dengan kolom 'BMI Status' tambahan.
    """
    # Membuat salinan DataFrame agar tidak memodifikasi DataFrame asli.
    df = dataframe.copy()

    # Menentukan status BMI berdasarkan kategori yang Anda inginkan.
    def bmi_status(bmi):
        if bmi < 18.5:
            return 'Underweight'
        elif 18.5 <= bmi < 24.9:
            return 'Normal'
        elif 25 <= bmi < 29.9:
            return 'Overweight'
        else:
            return 'Obesity'

    # Menambahkan kolom 'BMI Status' dengan memanggil fungsi bmi_status.
    df['BMI Status'] = df['BMI'].apply(bmi_status)

    return df

In [ ]:
#Memanggil dan menjalankan function penambahan status BMI yang sudah dideklarasikan

df_BMI_status = add_bmi_status(df_BMI)

df_BMI_New = pd.DataFrame(df_BMI_status)
df_BMI_New

nomor_peserta                         nama                      jurusan  \
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
11        DS5E12              Renata Havresia                   Akuntansi    
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (m)        BMI   BMI Status  
0                 79              1.72  26.703624   Overweight  
1                 49              1.53  20.932120       Normal  
2                 41              1.56  16.847469  Underweight  
3                 56              1.58  22.432303       Normal  
4                 60              1.73  20.047446       Normal  
5                 69              1.68  24.447279       Normal  
6                 57              1.62  21.719250       Normal  
7                 49              1.62  18.670934       Normal  
8                 58              1.67  20.796730       Normal  
9                 50              1.58  20.028842       Normal  
10                50              1.69  17.506390  Underweight  
11                63              1.59  24.919900      Obesity  
12                54              1.56  22.189349       Normal  
13                45              1.50  20.000000       Normal  
14                48              1.50  21.333333       Normal  
15                50              1.62  19.051974       Normal  
16                50              1.65  18.365473  Underweight  
17                58              1.68  20.549887       Normal  
18                40              1.65  14.692378  Underweight  
19                45              1.60  17.578125  Underweight  
20                52              1.65  19.100092       Normal  
21                54              1.60  21.093750       Normal  
22                53              1.59  20.964361       Normal

# **Mengurutkan Data**

1. Mengurutkan berdasarkan tinggi badan dari yang tertinggi ke rendah

In [ ]:
df_sorted_TB = df_BMI_New.sort_values(by='tinggi_badan (m)', ascending=False)
df_sorted_TB

nomor_peserta                         nama                      jurusan  \
4         DS5E05                    Mustafian                Ilmu Aktuaria   
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
17        DS5E18               Muhammad Wahyu                   Matematika   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
20        DS5E21               Zulfa Hafianah                    Manajemen   
18        DS5E19             Elina Salsabilah                  Sains Data    
16        DS5E17                Zahra Sevilla                   Matematika   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
11        DS5E12              Renata Havresia                   Akuntansi    
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   
9         DS5E10               Sylviana Mulia                    Geofisika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
12        DS5E13        Nova Masripah Maulida                   Matematika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   

    berat_badan (kg)  tinggi_badan (m)        BMI   BMI Status  
4                 60              1.73  20.047446       Normal  
0                 79              1.72  26.703624   Overweight  
10                50              1.69  17.506390  Underweight  
5                 69              1.68  24.447279       Normal  
17                58              1.68  20.549887       Normal  
8                 58              1.67  20.796730       Normal  
20                52              1.65  19.100092       Normal  
18                40              1.65  14.692378  Underweight  
16                50              1.65  18.365473  Underweight  
15                50              1.62  19.051974       Normal  
6                 57              1.62  21.719250       Normal  
7                 49              1.62  18.670934       Normal  
21                54              1.60  21.093750       Normal  
19                45              1.60  17.578125  Underweight  
11                63              1.59  24.919900      Obesity  
22                53              1.59  20.964361       Normal  
9                 50              1.58  20.028842       Normal  
3                 56              1.58  22.432303       Normal  
12                54              1.56  22.189349       Normal  
2                 41              1.56  16.847469  Underweight  
1                 49              1.53  20.932120       Normal  
14                48              1.50  21.333333       Normal  
13                45              1.50  20.000000       Normal

2. Mengurutkan Berdasarkan Nama dari Z - A

In [ ]:
df_sorted_Nama = df_BMI_New.sort_values(by='nama', ascending=False)
df_sorted_Nama

nomor_peserta                         nama                      jurusan  \
20        DS5E21               Zulfa Hafianah                    Manajemen   
16        DS5E17                Zahra Sevilla                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
9         DS5E10               Sylviana Mulia                    Geofisika   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
11        DS5E12              Renata Havresia                   Akuntansi    
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
12        DS5E13        Nova Masripah Maulida                   Matematika   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   
4         DS5E05                    Mustafian                Ilmu Aktuaria   
17        DS5E18               Muhammad Wahyu                   Matematika   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
18        DS5E19             Elina Salsabilah                  Sains Data    
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
5         DS5E06              Daniel Manurung                Ilmu Komputer   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
0         DS5E01       Adam Ibnudzaky Handoko                   Statistika   

    berat_badan (kg)  tinggi_badan (m)        BMI   BMI Status  
20                52              1.65  19.100092       Normal  
16                50              1.65  18.365473  Underweight  
13                45              1.50  20.000000       Normal  
9                 50              1.58  20.028842       Normal  
15                50              1.62  19.051974       Normal  
19                45              1.60  17.578125  Underweight  
11                63              1.59  24.919900      Obesity  
21                54              1.60  21.093750       Normal  
12                54              1.56  22.189349       Normal  
22                53              1.59  20.964361       Normal  
4                 60              1.73  20.047446       Normal  
17                58              1.68  20.549887       Normal  
6                 57              1.62  21.719250       Normal  
10                50              1.69  17.506390  Underweight  
8                 58              1.67  20.796730       Normal  
14                48              1.50  21.333333       Normal  
1                 49              1.53  20.932120       Normal  
18                40              1.65  14.692378  Underweight  
3                 56              1.58  22.432303       Normal  
5                 69              1.68  24.447279       Normal  
2                 41              1.56  16.847469  Underweight  
7                 49              1.62  18.670934       Normal  
0                 79              1.72  26.703624   Overweight

#**Filtering**

1. Berdasarkan Status BMI di Luar Normal

In [ ]:
filtered_df_BMI = df_BMI_New[df_BMI_New["BMI Status"] != "Normal"]
filtered_df_BMI

nomor_peserta                      nama      jurusan  berat_badan (kg)  \
0         DS5E01    Adam Ibnudzaky Handoko   Statistika                79   
2         DS5E03  Astrid Dwi Monica Roseva   Matematika                41   
10        DS5E11        Irzan Rafi Imtinan  Informatika                50   
11        DS5E12           Renata Havresia   Akuntansi                 63   
16        DS5E17             Zahra Sevilla   Matematika                50   
18        DS5E19          Elina Salsabilah  Sains Data                 40   
19        DS5E20          Seli Ayu Pertiwi   Manajemen                 45   

    tinggi_badan (m)        BMI   BMI Status  
0               1.72  26.703624   Overweight  
2               1.56  16.847469  Underweight  
10              1.69  17.506390  Underweight  
11              1.59  24.919900      Obesity  
16              1.65  18.365473  Underweight  
18              1.65  14.692378  Underweight  
19              1.60  17.578125  Underweight

2. Berdasarkan Tinggi Badan > 169 cm dan Berat Badan > 60 kg

In [ ]:
filtered_df_TB_BB = df_BMI_New.loc[(df_BMI_New["tinggi_badan (m)"] > 1.69) & (df_BMI_New["berat_badan (kg)"] > 60)]
filtered_df_TB_BB

nomor_peserta                    nama     jurusan  berat_badan (kg)  \
0        DS5E01  Adam Ibnudzaky Handoko  Statistika                79   

   tinggi_badan (m)        BMI  BMI Status  
0              1.72  26.703624  Overweight

3. Berdasarkan Nama yang mengandung Huruf "y"

In [ ]:
filtered_df_Nama = df_BMI_New[df_BMI_New['nama'].str.contains('y', case=False)]
filtered_df_Nama

nomor_peserta                         nama      jurusan  berat_badan (kg)  \
0         DS5E01       Adam Ibnudzaky Handoko   Statistika                79   
6         DS5E07         Lisya Septyo Ningrum  Sains Data                 57   
7         DS5E08     Anisyabella Muhdyna Suci    Manajemen                49   
9         DS5E10               Sylviana Mulia    Geofisika                50   
17        DS5E18               Muhammad Wahyu   Matematika                58   
19        DS5E20             Seli Ayu Pertiwi   Manajemen                 45   
22        DS5E23  Naurah Tsabitah Dzahabiyyah   Matematika                53   

    tinggi_badan (m)        BMI   BMI Status  
0               1.72  26.703624   Overweight  
6               1.62  21.719250       Normal  
7               1.62  18.670934       Normal  
9               1.58  20.028842       Normal  
17              1.68  20.549887       Normal  
19              1.60  17.578125  Underweight  
22              1.59  20.964361       Normal

4. Berdasarkan Berat Badan < 60 kg

In [ ]:
filtered_df_BB = df_BMI_New.loc[df_BMI_New['berat_badan (kg)'] < 60]
filtered_df_BB

nomor_peserta                         nama                      jurusan  \
1         DS5E02       Hamida Wefi Niamaputri                  Informatika   
2         DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3         DS5E04           Della Puspita Sari             Proteksi Tanaman   
6         DS5E07         Lisya Septyo Ningrum                  Sains Data    
7         DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8         DS5E09        Irza Dwi Niesviantika                  Informatika   
9         DS5E10               Sylviana Mulia                    Geofisika   
10        DS5E11           Irzan Rafi Imtinan                  Informatika   
12        DS5E13        Nova Masripah Maulida                   Matematika   
13        DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14        DS5E15          Herliza Khairunnisa                    Akuntansi   
15        DS5E16          Septia Kurrota Aini                   Manajemen    
16        DS5E17                Zahra Sevilla                   Matematika   
17        DS5E18               Muhammad Wahyu                   Matematika   
18        DS5E19             Elina Salsabilah                  Sains Data    
19        DS5E20             Seli Ayu Pertiwi                   Manajemen    
20        DS5E21               Zulfa Hafianah                    Manajemen   
21        DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22        DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    berat_badan (kg)  tinggi_badan (m)        BMI   BMI Status  
1                 49              1.53  20.932120       Normal  
2                 41              1.56  16.847469  Underweight  
3                 56              1.58  22.432303       Normal  
6                 57              1.62  21.719250       Normal  
7                 49              1.62  18.670934       Normal  
8                 58              1.67  20.796730       Normal  
9                 50              1.58  20.028842       Normal  
10                50              1.69  17.506390  Underweight  
12                54              1.56  22.189349       Normal  
13                45              1.50  20.000000       Normal  
14                48              1.50  21.333333       Normal  
15                50              1.62  19.051974       Normal  
16                50              1.65  18.365473  Underweight  
17                58              1.68  20.549887       Normal  
18                40              1.65  14.692378  Underweight  
19                45              1.60  17.578125  Underweight  
20                52              1.65  19.100092       Normal  
21                54              1.60  21.093750       Normal  
22                53              1.59  20.964361       Normal

# **Menambahkan Nomor Urut**

In [ ]:
# Membuat DataFrame baru untuk nomor urut
nomor_urut = pd.DataFrame({'No.': range(1, len(df) + 1)})

# Menggabungkan DataFrame nomor_urut dengan DataFrame asli
df_New = pd.concat([nomor_urut, df_BMI_New], axis=1)
df_New

No. nomor_peserta                         nama  \
0     1        DS5E01       Adam Ibnudzaky Handoko   
1     2        DS5E02       Hamida Wefi Niamaputri   
2     3        DS5E03     Astrid Dwi Monica Roseva   
3     4        DS5E04           Della Puspita Sari   
4     5        DS5E05                    Mustafian   
5     6        DS5E06              Daniel Manurung   
6     7        DS5E07         Lisya Septyo Ningrum   
7     8        DS5E08     Anisyabella Muhdyna Suci   
8     9        DS5E09        Irza Dwi Niesviantika   
9    10        DS5E10               Sylviana Mulia   
10   11        DS5E11           Irzan Rafi Imtinan   
11   12        DS5E12              Renata Havresia   
12   13        DS5E13        Nova Masripah Maulida   
13   14        DS5E14                   Wulan Sari   
14   15        DS5E15          Herliza Khairunnisa   
15   16        DS5E16          Septia Kurrota Aini   
16   17        DS5E17                Zahra Sevilla   
17   18        DS5E18               Muhammad Wahyu   
18   19        DS5E19             Elina Salsabilah   
19   20        DS5E20             Seli Ayu Pertiwi   
20   21        DS5E21               Zulfa Hafianah   
21   22        DS5E22            Nunung Novitasari   
22   23        DS5E23  Naurah Tsabitah Dzahabiyyah   

                        jurusan  berat_badan (kg)  tinggi_badan (m)  \
0                    Statistika                79              1.72   
1                   Informatika                49              1.53   
2                    Matematika                41              1.56   
3              Proteksi Tanaman                56              1.58   
4                 Ilmu Aktuaria                60              1.73   
5                 Ilmu Komputer                69              1.68   
6                   Sains Data                 57              1.62   
7                     Manajemen                49              1.62   
8                   Informatika                58              1.67   
9                     Geofisika                50              1.58   
10                  Informatika                50              1.69   
11                   Akuntansi                 63              1.59   
12                   Matematika                54              1.56   
13  Bahasa dan Sastra Indonesia                45              1.50   
14                    Akuntansi                48              1.50   
15                   Manajemen                 50              1.62   
16                   Matematika                50              1.65   
17                   Matematika                58              1.68   
18                  Sains Data                 40              1.65   
19                   Manajemen                 45              1.60   
20                    Manajemen                52              1.65   
21                Ilmu Aktuaria                54              1.60   
22                   Matematika                53              1.59   

          BMI   BMI Status  
0   26.703624   Overweight  
1   20.932120       Normal  
2   16.847469  Underweight  
3   22.432303       Normal  
4   20.047446       Normal  
5   24.447279       Normal  
6   21.719250       Normal  
7   18.670934       Normal  
8   20.796730       Normal  
9   20.028842       Normal  
10  17.506390  Underweight  
11  24.919900      Obesity  
12  22.189349       Normal  
13  20.000000       Normal  
14  21.333333       Normal  
15  19.051974       Normal  
16  18.365473  Underweight  
17  20.549887       Normal  
18  14.692378  Underweight  
19  17.578125  Underweight  
20  19.100092       Normal  
21  21.093750       Normal  
22  20.964361       Normal

In [ ]:
#Merapikan header agar lebih rapi dan enak dipandang

new_columns = ['No.', 'Id_Peserta', 'Nama', 'Jurusan', 'Berat_Badan (Kg)', 'Tinggi_Badan (M)', 'BMI', 'Status']
df_New.columns = new_columns
df_New.columns

Index(['No.', 'Id_Peserta', 'Nama', 'Jurusan', 'Berat_Badan (Kg)',
       'Tinggi_Badan (M)', 'BMI', 'Status'],
      dtype='object')

In [ ]:
df_New

No. Id_Peserta                         Nama                      Jurusan  \
0     1     DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1     2     DS5E02       Hamida Wefi Niamaputri                  Informatika   
2     3     DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3     4     DS5E04           Della Puspita Sari             Proteksi Tanaman   
4     5     DS5E05                    Mustafian                Ilmu Aktuaria   
5     6     DS5E06              Daniel Manurung                Ilmu Komputer   
6     7     DS5E07         Lisya Septyo Ningrum                  Sains Data    
7     8     DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8     9     DS5E09        Irza Dwi Niesviantika                  Informatika   
9    10     DS5E10               Sylviana Mulia                    Geofisika   
10   11     DS5E11           Irzan Rafi Imtinan                  Informatika   
11   12     DS5E12              Renata Havresia                   Akuntansi    
12   13     DS5E13        Nova Masripah Maulida                   Matematika   
13   14     DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14   15     DS5E15          Herliza Khairunnisa                    Akuntansi   
15   16     DS5E16          Septia Kurrota Aini                   Manajemen    
16   17     DS5E17                Zahra Sevilla                   Matematika   
17   18     DS5E18               Muhammad Wahyu                   Matematika   
18   19     DS5E19             Elina Salsabilah                  Sains Data    
19   20     DS5E20             Seli Ayu Pertiwi                   Manajemen    
20   21     DS5E21               Zulfa Hafianah                    Manajemen   
21   22     DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22   23     DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    Berat_Badan (Kg)  Tinggi_Badan (M)        BMI       Status  
0                 79              1.72  26.703624   Overweight  
1                 49              1.53  20.932120       Normal  
2                 41              1.56  16.847469  Underweight  
3                 56              1.58  22.432303       Normal  
4                 60              1.73  20.047446       Normal  
5                 69              1.68  24.447279       Normal  
6                 57              1.62  21.719250       Normal  
7                 49              1.62  18.670934       Normal  
8                 58              1.67  20.796730       Normal  
9                 50              1.58  20.028842       Normal  
10                50              1.69  17.506390  Underweight  
11                63              1.59  24.919900      Obesity  
12                54              1.56  22.189349       Normal  
13                45              1.50  20.000000       Normal  
14                48              1.50  21.333333       Normal  
15                50              1.62  19.051974       Normal  
16                50              1.65  18.365473  Underweight  
17                58              1.68  20.549887       Normal  
18                40              1.65  14.692378  Underweight  
19                45              1.60  17.578125  Underweight  
20                52              1.65  19.100092       Normal  
21                54              1.60  21.093750       Normal  
22                53              1.59  20.964361       Normal

# **Mencari Nilai-Nilai Berikut**

1. Kelompokkan Tinggi badan dan berat Badan, serta cari nilai terendah dan tertingginya.

Berat Badan

In [ ]:
df_BB = df_New.drop(['Tinggi_Badan (M)', 'BMI', 'Status'], axis=1)
df_BB

No. Id_Peserta                         Nama                      Jurusan  \
0     1     DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1     2     DS5E02       Hamida Wefi Niamaputri                  Informatika   
2     3     DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3     4     DS5E04           Della Puspita Sari             Proteksi Tanaman   
4     5     DS5E05                    Mustafian                Ilmu Aktuaria   
5     6     DS5E06              Daniel Manurung                Ilmu Komputer   
6     7     DS5E07         Lisya Septyo Ningrum                  Sains Data    
7     8     DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8     9     DS5E09        Irza Dwi Niesviantika                  Informatika   
9    10     DS5E10               Sylviana Mulia                    Geofisika   
10   11     DS5E11           Irzan Rafi Imtinan                  Informatika   
11   12     DS5E12              Renata Havresia                   Akuntansi    
12   13     DS5E13        Nova Masripah Maulida                   Matematika   
13   14     DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14   15     DS5E15          Herliza Khairunnisa                    Akuntansi   
15   16     DS5E16          Septia Kurrota Aini                   Manajemen    
16   17     DS5E17                Zahra Sevilla                   Matematika   
17   18     DS5E18               Muhammad Wahyu                   Matematika   
18   19     DS5E19             Elina Salsabilah                  Sains Data    
19   20     DS5E20             Seli Ayu Pertiwi                   Manajemen    
20   21     DS5E21               Zulfa Hafianah                    Manajemen   
21   22     DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22   23     DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    Berat_Badan (Kg)  
0                 79  
1                 49  
2                 41  
3                 56  
4                 60  
5                 69  
6                 57  
7                 49  
8                 58  
9                 50  
10                50  
11                63  
12                54  
13                45  
14                48  
15                50  
16                50  
17                58  
18                40  
19                45  
20                52  
21                54  
22                53

In [ ]:
# Mencari nilai terendah (minimum) dan tertinggi (maksimum) dari kolom 'Berat_Badan (Kg)'
min_BB = df_BB['Berat_Badan (Kg)'].min()
max_BB = df_BB['Berat_Badan (Kg)'].max()

print("Nilai Terendah Berat Badan: ", min_BB, "Kg")
print("Nilai Tertinggi Berat Badan: ", max_BB, "Kg")

Nilai Terendah Berat Badan:  40 Kg
Nilai Tertinggi Berat Badan:  79 Kg


Tinggi Badan

In [ ]:
df_TB = df_New.drop(['Berat_Badan (Kg)', 'BMI', 'Status'], axis=1)
df_TB

No. Id_Peserta                         Nama                      Jurusan  \
0     1     DS5E01       Adam Ibnudzaky Handoko                   Statistika   
1     2     DS5E02       Hamida Wefi Niamaputri                  Informatika   
2     3     DS5E03     Astrid Dwi Monica Roseva                   Matematika   
3     4     DS5E04           Della Puspita Sari             Proteksi Tanaman   
4     5     DS5E05                    Mustafian                Ilmu Aktuaria   
5     6     DS5E06              Daniel Manurung                Ilmu Komputer   
6     7     DS5E07         Lisya Septyo Ningrum                  Sains Data    
7     8     DS5E08     Anisyabella Muhdyna Suci                    Manajemen   
8     9     DS5E09        Irza Dwi Niesviantika                  Informatika   
9    10     DS5E10               Sylviana Mulia                    Geofisika   
10   11     DS5E11           Irzan Rafi Imtinan                  Informatika   
11   12     DS5E12              Renata Havresia                   Akuntansi    
12   13     DS5E13        Nova Masripah Maulida                   Matematika   
13   14     DS5E14                   Wulan Sari  Bahasa dan Sastra Indonesia   
14   15     DS5E15          Herliza Khairunnisa                    Akuntansi   
15   16     DS5E16          Septia Kurrota Aini                   Manajemen    
16   17     DS5E17                Zahra Sevilla                   Matematika   
17   18     DS5E18               Muhammad Wahyu                   Matematika   
18   19     DS5E19             Elina Salsabilah                  Sains Data    
19   20     DS5E20             Seli Ayu Pertiwi                   Manajemen    
20   21     DS5E21               Zulfa Hafianah                    Manajemen   
21   22     DS5E22            Nunung Novitasari                Ilmu Aktuaria   
22   23     DS5E23  Naurah Tsabitah Dzahabiyyah                   Matematika   

    Tinggi_Badan (M)  
0               1.72  
1               1.53  
2               1.56  
3               1.58  
4               1.73  
5               1.68  
6               1.62  
7               1.62  
8               1.67  
9               1.58  
10              1.69  
11              1.59  
12              1.56  
13              1.50  
14              1.50  
15              1.62  
16              1.65  
17              1.68  
18              1.65  
19              1.60  
20              1.65  
21              1.60  
22              1.59

In [ ]:
# Mencari nilai terendah (minimum) dan tertinggi (maksimum) dari kolom 'Tinggi_Badan (M)'
min_TB = df_TB['Tinggi_Badan (M)'].min()
max_TB = df_TB['Tinggi_Badan (M)'].max()

print("Nilai Terendah Berat Badan: ", min_TB, "M")
print("Nilai Tertinggi Berat Badan: ", max_TB, "M")

Nilai Terendah Berat Badan:  1.5 M
Nilai Tertinggi Berat Badan:  1.73 M


2. Rata-Rata Tinggi Badan

In [ ]:
# Mencari rata-rata tinggi badan dalam kolom 'Tinggi_Badan (M)'
average_TB = df_TB['Tinggi_Badan (M)'].mean().round(2)    #Fungsi Round(2) berguna untuk melakukan pembulatan 2 angka dibelakang koma

print("Rata-rata Tinggi Badan: ", average_TB, "M")

Rata-rata Tinggi Badan:  1.62 M


3. Rata-Rata Berat Badan

In [ ]:
# Mencari rata-rata berat badan dalam kolom 'Berat_Badan (Kg)'
average_BB = df_BB['Berat_Badan (Kg)'].mean().round(2)    #Fungsi Round(2) berguna untuk melakukan pembulatan 2 angka dibelakang koma

print("Rata-rata Berat Badan: ", average_BB, "Kg")

Rata-rata Berat Badan:  53.48 Kg


4. banyaknya orang atau peserta yang tercatat pada DataFrame

In [ ]:
# Menghitung banyak data pada DataFrame melalui kolom Id_Peserta
banyak_data = df_New['Id_Peserta'].count()

print("Banyak Peserta yang tercatat : ", banyak_data, "Orang")

Banyak Peserta yang tercatat :  23 Orang


5. Orang dengan nilai BMI Tertinggi

In [ ]:
# Menggunakan fungsi nlargest untuk mendapatkan 5 orang dengan nilai BMI tertinggi
top_5_BMI = df_New.nlargest(5, 'BMI')
top_5_BMI

No. Id_Peserta                    Nama           Jurusan  \
0     1     DS5E01  Adam Ibnudzaky Handoko        Statistika   
11   12     DS5E12         Renata Havresia        Akuntansi    
5     6     DS5E06         Daniel Manurung     Ilmu Komputer   
3     4     DS5E04      Della Puspita Sari  Proteksi Tanaman   
12   13     DS5E13   Nova Masripah Maulida        Matematika   

    Berat_Badan (Kg)  Tinggi_Badan (M)        BMI      Status  
0                 79              1.72  26.703624  Overweight  
11                63              1.59  24.919900     Obesity  
5                 69              1.68  24.447279      Normal  
3                 56              1.58  22.432303      Normal  
12                54              1.56  22.189349      Normal

6. Banyaknya orang di setiap status BMI (Underweight, Normal, Overweight, dan Obesity)

In [ ]:
#Menghitung banyaknya orang di setiap status BMI
banyak_status_BMI = df_New['Status'].value_counts()

banyak_status_BMI

Normal         16
Underweight     5
Overweight      1
Obesity         1
Name: Status, dtype: int64